modules/api/php/endpoints/project/dicoms.class.inc

```diff
@@ -0,0 +1,153 @@
+ * @link     https://github.com/aces/Loris
+ */
+class Dicoms extends Endpoint implements \LORIS\Middleware\ETagCalculator
+{
+    /**
+     * A cache of the results of the endpoint, so that
+     * it doesn't need to be recalculated for the ETag and handler
+     */
+    private $_cache;
+
+    /**
+     * The requested project
+     */
+    private $_project;
+
+    /**
+     * Contructor
+     *
+     * @param \Project $project The requested project
+     */
+    public function __construct(\Project $project)
+    {
+        $this->_project = $project;
+    }
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return ['GET'];
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return ["v0.0.3"];
+    }
+
+    /**
+     * Handles a request that starts with /projects/$projectname/candidates
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $pathparts = $request->getAttribute('pathparts');
+        if (count($pathparts) !== 0) {
+            return new \LORIS\Http\Response\JSON\NotFound();
+        }
+
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+
+        case 'OPTIONS':
+            return (new \LORIS\Http\Response())
+                ->withHeader('Allow', $this->allowedMethods());
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Create an array representation of this endpoint's reponse body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        if (isset($this->_cache)) {
+            return $this->_cache;
+        }
+
+        try {
+            $datestring = $request->getQueryParams()['since'] ?? '1970-01-01';
+            $since      = new \DateTime($datestring);
+        } catch (\Exception $e) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                $e->getMessage()
+            );
+        }
+
+        $provisioner = new \LORIS\api\Provisioners\ProjectDicomsObjectProvisioner(
+            $this->_project,
+            $since,
+            '\LORIS\api\Models\ProjectDicomsObject'
+        );
+
+        $all = $provisioner->getAllInstances();
```

Comment:
`$all` ?? all what?

Reply:
`$provisioner->getAllInstances` , so obviously all instances (of dicoms object). I get the point you are making though, I'll give it a more meaningful name, like `$allDicomsInstances`

Reply:
now that I'm using `iterator_to_array` as you suggested, I don't think this variable is necessary anymore, I did `$dicoms = iterator_to_array($provisioner->getAllInstances());`, which is consistent with the others endpoints (e.g. for [images](https://github.com/aces/Loris/blob/a87ead08684d8f4be4e10a982339eae440b075e7/modules/api/php/endpoints/project/images.class.inc#L131), the final array containing all info to display about images in a project is called `$images` )